spec/challenge.spec.js

```diff
@@ -1,127 +1,104 @@
-        expect(this.fourthDequeued).to.deep.eq(this.secondCargoPlane);
-      });
-    });
-  });
-});
+'use strict'
+
+const chai = require('chai')
+const chaiAsPromised = require('chai-as-promised')
+
+chai.use(chaiAsPromised)
+
+const expect = chai.expect
+
+const ATCQueue = require('../lib/challenge')
+
+const passengerLarge = { type: 'passenger', size: 'large' }
+const passengerSmall = { type: 'passenger', size: 'small' }
+const cargoLarge = { type: 'cargo', size: 'large' }
+const cargoSmall = { type: 'cargo', size: 'small' }
+const aircrafts = [passengerLarge, passengerSmall, cargoLarge, cargoSmall]
+
+describe('ATCQueue', () => {
+  beforeEach(() => {
+    this.atcQueue = new ATCQueue()
+  })
+
+  describe('aircraftCount()', () => {
+    it('returns 0 if the aircraft queue is empty', () => {
+      expect(this.atcQueue.aircraftCount()).to.equal(0)
+    })
+  })
+
+  describe('enqueue()', () => {
+    it('adds aircrafts to the queue', () => {
+      aircrafts.forEach((aircraft, index) => {
+        this.atcQueue.enqueue(aircraft)
+
+        expect(this.atcQueue.aircraftCount()).to.equal(index + 1)
+      })
+    })
+  })
+
+  describe('dequeue()', () => {
+    it('removes aircrafts from the queue', () => {
+      aircrafts.forEach(aircraft => this.atcQueue.enqueue(aircraft))
+
+      for (let i = 1; i <= aircrafts.length; i++) {
+        this.atcQueue.dequeue()
+
+        expect(this.atcQueue.aircraftCount()).to.equal(aircrafts.length - i)
+      }
+    })
+
+    it('removes passenger aircrafts before cargo aircrafts', () => {
+      this.atcQueue.enqueue(cargoSmall)
+      this.atcQueue.enqueue(passengerSmall)
+      this.atcQueue.enqueue(passengerLarge)
+      this.atcQueue.enqueue(cargoLarge)
+
+      expect(this.atcQueue.dequeue()).to.have.property('type', 'passenger')
+      expect(this.atcQueue.dequeue()).to.have.property('type', 'passenger')
+      expect(this.atcQueue.dequeue()).to.have.property('type', 'cargo')
+      expect(this.atcQueue.dequeue()).to.have.property('type', 'cargo')
+    })
+
+    it('removes large aircrafts before small aircrafts of the same type', () => {
+      this.atcQueue.enqueue(cargoSmall)
+      this.atcQueue.enqueue(passengerSmall)
+      this.atcQueue.enqueue(passengerLarge)
+      this.atcQueue.enqueue(cargoLarge)
+
+      expect(this.atcQueue.dequeue()).to.deep.equal(passengerLarge)
+      expect(this.atcQueue.dequeue()).to.deep.equal(passengerSmall)
+      expect(this.atcQueue.dequeue()).to.deep.equal(cargoLarge)
+      expect(this.atcQueue.dequeue()).to.deep.equal(cargoSmall)
+    })
+
+    it('removes earlier enqueued aircrafts of the same type and size first', () => {
+      const firstPassengerLarge = Object.assign({ order: 1 }, passengerLarge)
+      const secondPassengerLarge = Object.assign({ order: 2 }, passengerLarge)
+      const firstPassengerSmall = Object.assign({ order: 1 }, passengerSmall)
+      const secondPassengerSmall = Object.assign({ order: 2 }, passengerSmall)
```

Comment:
I think with the changes you made the order property here is unnecessary. (Or it was unnecessary beforehand. Either way, it can go.)